test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsSolutionRestoreServiceTests.cs

```diff
@@ -1716,8 +1716,43 @@ public void NominateProjectAsync_ThrowsNullReferenceException()
             _ = Assert.ThrowsAsync<ArgumentNullException>(async () => await service.NominateProjectAsync(@"F:\project\project.csproj", (IVsProjectRestoreInfo)null, CancellationToken.None));
 
             _ = Assert.ThrowsAsync<ArgumentNullException>(async () => await service.NominateProjectAsync(@"F:\project\project.csproj", (IVsProjectRestoreInfo2)null, CancellationToken.None));
+        }
+
+        [Fact]
+        public async Task NominateProjectAsync_InvalidTargetFrameworkMoniker_DoesNotThrow()
+        {
+            // Arrange
+            var cache = new Mock<IProjectSystemCache>();
+            cache.Setup(x => x.AddProjectRestoreInfo(
+                    It.IsAny<ProjectNames>(),
+                    It.IsAny<DependencyGraphSpec>()))
+                .Returns(true);
 
+            var restoreWorker = new Mock<ISolutionRestoreWorker>();
 
+            var logger = new Mock<ILogger>();
+
+            var service = new VsSolutionRestoreService(cache.Object, restoreWorker.Object, logger.Object);
+
+            var emptyReferenceItems = Array.Empty<VsReferenceItem>();
+            var projectRestoreInfo = new VsProjectRestoreInfo2(@"f:\project\",
+                new VsTargetFrameworks2(new []
+                {
+                    new VsTargetFrameworkInfo2(
+                        targetFrameworkMoniker: "_,Version=2.0",
+                        packageReferences: emptyReferenceItems,
+                        projectReferences: emptyReferenceItems,
+                        packageDownloads: emptyReferenceItems,
+                        frameworkReferences: emptyReferenceItems,
+                        projectProperties: Array.Empty<IVsProjectProperty>())
+                }));
+
+            // Act
+            var result = await service.NominateProjectAsync(@"f:\project\project.csproj", projectRestoreInfo, CancellationToken.None);
+
+            // Assert
+            Assert.False(result);
```

Comment:
I like this test, as it seems well organized and easy to understand, even though I'm new to many of the classes it's using.
One thing I wonder is if an exception is thrown, are you relying on the exception to bubble out to the testing framework? Would it be better to catch exceptions, and Assert.Fail() if any are caught? That way you can specify a friendly/useful test failure message.

Reply:
Actually, I realize now that `NominateProjectAsync` catches exceptions. So perhaps my only suggestion would be to rename this method `DoesNotThrow` to something else (`ReturnsFalseWhenInvalid` ??). It's actually not testing for `throw`ing, which is what confused me.